### ISSUE TITLE
Fix: Tinyman price can lead to incorrect orders in algodex orderbook

### ISSUE THREAD

**Issue by ericsharma** [algodex/trading-bot#159]
# ℹ Overview
When the price of an asset on tinyman is different than the bid/ask spread on the algodex orderbook, maker orders are placed when taker execution should occur. 

### USDC/Algo orderbook before running the bot 
<img width="309" alt="Screen Shot 2022-11-17 at 10 57 54 AM" src="https://user-images.githubusercontent.com/26069470/202495276-ad8dce38-fd8e-461e-a220-2e5b0b945478.png">

### USDC/Algo bot settings
<img width="1591" alt="Screen Shot 2022-11-17 at 11 04 16 AM" src="https://user-images.githubusercontent.com/26069470/202496761-80f7e7c5-0894-45b5-b1ee-7f35d139c460.png">

### USDC/Algo orderbook after running the bot
<img width="311" alt="Screen Shot 2022-11-17 at 11 05 45 AM" src="https://user-images.githubusercontent.com/26069470/202497989-e13d0b44-f530-4715-ab0a-d2ba038ccad4.png">

The bot adds a buy order at a price higher than an existing sell order in our orderbook which breaks the expected behavior of the orderbook.

## Proposed solution
Fetch the algodex orderbook for the targret asset and prevent the bot from running when the above conditions are met.
### The above behavior is an arbitrage opportunity
We can have a general message template that alerts the user of the pricing mismatch and how best to exploit it. 
So given the example above, instead of placing the order we would alert the user with a message: 
``` 
You can currently buy USDC for 0.89 algo on Algodex and you can sell USDC for 1.3766 algo on tinman. Our liquidity bot can only operate when the market price is in equilibrium 

```

**Comment by atrefonas** [comment]
This also isn’t a problem on Mainnet since other people actively exploit
arb opportunities.

**Comment by ericsharma** [comment]
Agreed that these types of price discrepancies are much rarer on main-net and going back to your point about exploiting arbitrage opportunities as taker orders: In the above example, you are saying that instead of placing the algo buy order ithe bot should have executed on the cheaper sell order right?

**Comment by ericsharma** [comment]
I tested it with execution of both instead of masker and it does execute on the better orderbook deal but then another issue arises:
- The bot places both legs of the trade using both execution
- one leg executes while the other stays open
-  the bot see's that only one leg is open so it replaces the missing leg at the current tinyman price
- the cycle repeats. 
As a consequence, the user will end up incrementally buying the open order on the favorable leg which will probably be surprising without some type of message

**Comment by atrefonas** [comment]
I think let's just save this for the V2. Like I mentioned this whole thing
isn't a real issue on Mainnet because other people are running their own
bots that actively do arbitrage.
